clippy_lints/src/question_mark.rs

```diff
@@ -156,10 +235,32 @@ impl QuestionMark {
         }
     }
 
-    fn expression_returns_unmodified_err(cx: &LateContext<'_>, expr: &Expr<'_>, cond_expr: &Expr<'_>) -> bool {
-        match peel_blocks_with_stmt(expr).kind {
-            ExprKind::Ret(Some(ret_expr)) => Self::expression_returns_unmodified_err(cx, ret_expr, cond_expr),
+    fn expression_returns_unmodified_err(
+        cx: &LateContext<'_>,
+        expr: &Expr<'_>,
+        cond_expr: &Expr<'_>,
+        pat_symbol: Option<Symbol>,
+    ) -> bool {
+        match &peel_blocks_with_stmt(expr).kind {
+            ExprKind::Ret(Some(ret_expr)) => {
+                Self::expression_returns_unmodified_err(cx, ret_expr, cond_expr, pat_symbol)
+            },
             ExprKind::Path(_) => path_to_local(expr).is_some() && path_to_local(expr) == path_to_local(cond_expr),
+            ExprKind::Call(call_expr, args_expr) => {
```

Comment:
This is no longer requiring `return ..`?

Reply:
Sorry, but I'm not sure I understand this, why "no longer requiring `return`?", can you explain this a bit more? XD

Reply:
Oh I thought this function is only supposed to return true for `return ..` expressions (`ExprKind::Ret`) but I guess that is not true. The name is just misleading.

Reply:
Oh I see~